Adds basic marks implementation

tdfacer opened the pull request:
### This is not quite ready to merge; I'm opening to facilitate discussion.  The code is working but the new tests are failing.
This seems to work well for setting/jumping to marks within the same file as well as into other files.  There are a few things I had questions on though:
* Would you recommend using another data structure instead of a `Dictionary`?  I was thinking `Map`s but couldn't get them working quite right.
* Would you recommend moving the `markDict` out of `Types.fs` and into the global state in the `Vim` module?
* The new unit tests fail with a `NullReferenceException`.  I think it may be due to improperly setting up the dictionary in the test runner, but I may need to play around with it a bit more.

nosami commented:
First of all, nice work! This looks like a great feature

> Would you recommend using another data structure instead of a Dictionary? I was thinking Maps but couldn't get them working quite right.

Yeah, a map would be more idiomatic. I can take a look if you want.

> Would you recommend moving the markDict out of Types.fs and into the global state in the Vim module?

I think so, yes. Right now, you have the markMap on the `VimState` record, which is an instance per editor. These need to be global. I'm not too bothered how you do this right now - I'll probably add a `GlobalState` record at some point in the future.

> The new unit tests fail with a NullReferenceException. I think it may be due to improperly setting up the dictionary in the test runner, but I may need to play around with it a bit more.

I haven't tested yet, but I would suspect that this would have a NRE at test time `IdeApp.Workbench.ActiveDocument.FileName.FullPath.ToString()`. A way to work around this might be to store the ActiveDocument on the vimstate record.

What about edits? They will cause the marks to move position. It might be worth trying to use the bookmarks feature of Visual Studio. (No idea whether or not that is possible using the public API)

nosami commented:
Also, it's not possible to test some things at all via unit tests. If you can't get the tests to pass but the code works, put the tests on a separate branch, and I can investigate later.

nosami commented:
Put `markMap` [here](https://github.com/nosami/XSVim/blob/4e95a10511cba66755a540b745f45f0e34e26e90/XSVim/XSVim.fs#L324). This will make it global across all documents.

It can stay as a Dictionary for now. I'll update the code at some future date when I handle global state in a nicer way.

tdfacer commented:
I made a little progress on this, but these issues still apply:
* `KeyNotFoundException` when trying to go to mark not in dict (see my many attempts to handle this☺️)
* The needed Bookmarks API does not seem to be public (unless I missed it). 
* Failing tests

What I fixed:
* Made marks a movement, so something like: `20Gmagg` followed by `v'a` (sub ` for ') would highlight the first 20 lines
* Moved `markDict` to requested location
* Moved failing tests to new branch

nosami commented:
I don't see a `KeyNotFoundException`... in fact, it seems to work perfectly :)

nosami commented:
I think this could work with the [Marker methods on editor](https://github.com/mono/monodevelop/blob/master/main/src/core/MonoDevelop.Ide/MonoDevelop.Ide.Editor/TextEditor.cs#L694-L764), but this would require quite a bit more work. E.g. I think the code would have to no longer rely on line numbers, but instead search a document for markers.

I'd be happy to merge this in now as it is, and it can be improved upon later

nosami commented:
Initial support for #68
